### DISCUSSION TITLE
Packages/com.unity.entities/manual/aspects-intro

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#547] [Notes]
# Packages/com.unity.entities/manual/aspects-intro

https://docs.unity3d.com/Packages/com.unity.entities@latest/index.html?subfolder=%2Fmanual%2Faspects-intro.html

**Comment by vertxxyz** [comment]
> Is there any overhead associated with having aspects? I assume they aren't free, will I run into issues if I have a lot of aspects?

[Dani](https://daxode.dk) (A Unity DOTS developer):
> Well, the cost of aspects are two fold. One is the runtime cost and the other is the iteration cost.  (that doesn't mean you can't use em, just know when to!)
>
> Let's start with the latter. Due to the nature of aspects, we have to do a lot of symbol analysis. This is expensive and so over time the more aspects you add the higher the chance of your project having loooong domain reload times. - So use with care. If you have like 50-100 aspects that shouldn't be much of an issue, the problem starts at higher scales.
>
> Now back to runtime cost. Aspects themselves don't do anything that a user couldn't have done themselves. And for the simple case might produce faster code, as they do cache themselves and are properly updated. They even work in jobs.
>
> The performance cost of Aspects is when you acquire to much access permission. Like say I have an aspect with 5 components, two of which are RW. Now I go in my job, but since every aspect is RW it means the two components now can't run parallel. Meaning the whole aspect has to block a previous job using the aspect to run the next.
>
> Note, this isn't actually a problem of the aspect itself, more the pattern that it lends itself to. E.g. for solving that problem still with aspect, you could have a `AttackDroneRO` and `AttackDroneRW`. If you're only using the RO version, it works on as many jobs you want until the `AttackDroneRW` job comes.
>
> That is to say. At runtime aspects have handwritten cached performance, but if you're not carefull with them, you might end up getting long job queues.
